aws/resource_aws_ses_template.go

```diff
@@ -0,0 +1,156 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ses"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsSesTemplate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsSesTemplateCreate,
+		Read:   resourceAwsSesTemplateRead,
+		Update: resourceAwsSesTemplateUpdate,
+		Delete: resourceAwsSesTemplateDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateSesTemplateName,
+			},
+			"html": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateSesTemplateHtml,
+			},
+			"subject": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"text": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateSesTemplateText,
+			},
+		},
+	}
+}
+func resourceAwsSesTemplateCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+
+	templateName := d.Get("name").(string)
+
+	template := ses.Template{
+		TemplateName: aws.String(templateName),
+	}
+
+	if v, ok := d.GetOk("html"); ok {
+		template.HtmlPart = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("subject"); ok {
+		template.SubjectPart = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("text"); ok {
+		template.TextPart = aws.String(v.(string))
+	}
+
+	input := ses.CreateTemplateInput{
+		Template: &template,
+	}
+
+	log.Printf("[DEBUG] Creating SES template: %#v", input)
+	_, err := conn.CreateTemplate(&input)
+	if err != nil {
+		return fmt.Errorf("Creating SES template failed: %s", err.Error())
+	}
+	d.SetId(templateName)
+
+	return resourceAwsSesTemplateRead(d, meta)
+}
+
+func resourceAwsSesTemplateRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).sesConn
+	input := ses.GetTemplateInput{
+		TemplateName: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Reading SES template: %#v", input)
+	gto, err := conn.GetTemplate(&input)
+	if err != nil {
+		if scErr, ok := err.(awserr.Error); ok && scErr.Code() == "TemplateDoesNotExist" {
```

Comment:
We have a nice helper function for this which can save us some characters + 1 import line.
```go
if isAWSErr(err, "TemplateDoesNotExist", "") {
```